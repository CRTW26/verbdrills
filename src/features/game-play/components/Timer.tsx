import React from 'react'
import { useTimer } from '../hooks/useTimer'

interface Props {
  duration: number
  onExpiration: () => void
}

const Timer: React.FC<Props> = ({ duration, onExpiration }) => {
  const { timeLeft, isTimerExpired, makeTimeString } = useTimer(
    duration * 60,
    onExpiration
  )

  return (
    <div>
      <h3>{isTimerExpired ? "Time's up!" : makeTimeString(timeLeft)}</h3>

      <style jsx>{`
        h3 {
          text-align: left;
        }
      `}</style>
    </div>
  )
}

export default Timer
