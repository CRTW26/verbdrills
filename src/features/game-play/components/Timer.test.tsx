import { render } from '@testing-library/react'
import Timer from './Timer'

describe('<Timer />', () => {
  test('should render', () => {
    const { asFragment } = render(
      <Timer timeRemaining={0} isTimerExpired={true} />
    )

    expect(asFragment).toMatchSnapshot()
  })
})
