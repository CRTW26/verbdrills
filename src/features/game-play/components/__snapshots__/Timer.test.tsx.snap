// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> should render 1`] = `
<body>
  <div>
    <div
      class="jsx-1645246723"
    >
      <h3
        class="jsx-1645246723"
      >
        1:00
      </h3>
    </div>
  </div>
</body>
`;
